src/index.ts

```diff
@@ -453,15 +506,21 @@ class Spanner extends GrpcService {
       },
       (err, operation, resp) => {
         if (err) {
-          callback(err, null, null, resp);
+          callback!(err, null, null, resp);
```

Comment:
If `callback` is undefined, I think here it will throw an error. 

```typescript
function createInstance(callback?: Function) {
    callback!("hello, world!");
}

createInstance();
```

This gives me an error: `VM52:3 Uncaught TypeError: callback is not a function`. 

From the [doc](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator), I feel like that `!` is used to mute the type checker instead of having any actual effect (stop the execution if the variable is undefined). 

I tried the following code as well: 

```typescript
interface IPerson {
   firstName: string;
   lastName: string;
}

function createInstance(person?: IPerson) {
    console.log(person!.firstName);
}

createInstance();
```

It throws an error: `VM58:6 Uncaught TypeError: Cannot read property 'firstName' of undefined`.

Reply:
the `callback` is optional from the user's point of view. If user omits the callback, a promise will be returned.

From the API's point of view the callback will always be there, it will be provided by `PromissifyAll`

Reply:
@AVaksman Thanks for explanation.